### ISSUE TITLE
2023.7: Error while setting up monitor_docker platform for switch

### ISSUE THREAD

**Issue by genem2** [ualex73/monitor_docker#128]
I'm guessing this might be an issue for this custom component, rather than HA, but let me know if that's not right.

With 2023.7, this now fails with the log entry shown below.
Wondering it it's related to their blog post regarding: [[Changes to entity naming](https://developers.home-assistant.io/blog/2023/06/28/entity-name-changes)]

```
2023-07-06 08:40:34.657 ERROR (MainThread) [homeassistant.components.switch] Error while setting up monitor_docker platform for switch
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 353, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/monitor_docker/switch.py", line 78, in async_setup_platform
    api = hass.data[DOMAIN][name][API]
          ~~~~~~~~~~~~~~~~~~~~~~~^^^^^
KeyError: 'api'
```

**Comment by ualex73** [comment]
I just upgraded my test HA to 2023.7.0, but it still works? Are you running it in docker or in another way?

**Comment by genem2** [comment]
Yes, it's in docker on my unraid server.

**Comment by genem2** [comment]
Well, appears to be resolved with 2023.7.1
